src/domain/navigation/index.js

```diff
@@ -132,11 +132,14 @@ export function parseUrlPath(urlPath, currentNavPath, defaultSessionId) {
                 segments.push(roomsSegmentWithRoom(rooms, roomId, currentNavPath));
             }
             segments.push(new Segment("room", roomId));
-            // Add right-panel segments from previous path
-            const previousSegments = currentNavPath.segments;
-            const i = previousSegments.findIndex(s => s.type === "right-panel");
-            if (i !== -1) {
-                segments.push(...previousSegments.slice(i));
+            if (!urlPath.includes("/member/")) {
```

Comment:
Hah, I completely misunderstood what you are doing here for some reason likely more related to me than to your code, to then look into the code wondering if we couldn't reuse the `open-room` action and come to exactly the same conclusion as you to then come back here and suggest that and go ... aaaaah, that's what's happening here :smile: 

Having said that, I wonder if we can make the check a bit more generic here by just checking if  there is only one element (the room id) after the `open-room` part in `parts`. Wdyt?

Reply:
Sounds good to me :+1: